### PR TITLE
NDVI automation files for peer review (2022)

### PR DESCRIPTION
**Pull request by S-Kur**
# Pull Request Template

## CI Checks
- [ ] The notebook runs from start to finish on all operating systems:
   - [ ] Mac
   - [ ] Windows
   - [ ] Linux 
### Reproducibility
- [ ] Are the data downloaded in the code
- [ ] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [ ] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [ ] Does the notebook run from start to finish?

### Functions
- [ ] Do functions follow PEP 8 format conventions?
- [ ] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [ ] Are function names expressive (the name describes what the function does)?
- [ ] Are functions easy to understand and read?
- [ ] How many tasks does each function do? (ideally a function does one thing well).
### Package imports
- [ ] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [ ] Are variable names throughout the code, expressive?
- [ ] Suggest changes if not, highlight what was done well
- [ ] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [ ] Are segments of code repeated in the file or is the code DRY?
- [ ] Are loops used to optimize DRY code?
- [ ] Are functions used to optimize DRY code?
- [ ] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [ ] Highlight any novel approaches to completing the assignment.

### CODE REVIEW

**Comment by Tsrockross** [comment]
Will make a start today on reviewing your code. Also going to wait until the lecture incase there are further instructions.

**Comment by Tsrockross** [comment]
[https://github.com/S-Kur/ea-2021-ndvi-automation-review/blob/2e1cf6a5fc3949206649445bfd4f3d2e60525c44/svetlana_kurakina_ndvi.py#L142](url)

File: svetlana_kurakina_ndvi.py [hunk 1]
```diff
@@ -0,0 +1,904 @@
...
+import numpy as np
+import pandas as  pd
+import geopandas as gpd
+import xarray as xr
+import rioxarray as rxr
+from rasterio.plot import plotting_extent
+import earthpy as et
+import earthpy.mask as em
+import earthpy.spatial as es
+import earthpy.plot as ep
```
Comment (Tsrockross) [inline svetlana_kurakina_ndvi.py:157]: Nice set of imports. Nothing unnecessary!

File: svetlana_kurakina_ndvi.py [hunk 2]
```diff
@@ -0,0 +1,904 @@
...
+# In this cell place all of the functions needed to run your notebook
+# You will be graded here on function application, docstrings, efficiency so ensure
+# All functions are placed here!
+
+# YOUR CODE HERE
+
+# Define function which opens, crops, and cleans landsat bands
+def open_clean_bands(band_path,
+                     crop_extent,
+                     valid_range=None,):
```
Comment (Tsrockross) [inline svetlana_kurakina_ndvi.py:327]: Should '=' have spaces around it? There are a few more instances I've spotted in the code.

File: svetlana_kurakina_ndvi.py [hunk 3]
```diff
@@ -0,0 +1,904 @@
...
+# Set figure
+fig, ax = plt.subplots(figsize=(12, 6))
+
+# Loop through dataframe and plot NDVI for each site using assigned color
+for idx, asite in enumerate(plot_sites):
+    temp = ndvi_clean[ndvi_clean['site']==asite]
+    ax.plot(temp.index,
+            temp.mean_ndvi,
+            label=plot_sites[idx],
+            color = plot_colors[idx])
```
Comment (Tsrockross) [inline svetlana_kurakina_ndvi.py:781]: I like this loop for plotting, a lot cleaner than the one I used!

File: svetlana_kurakina_ndvi.py [hunk 4]
```diff
@@ -0,0 +1,904 @@
...
+)
+
+# Export final df to csv
+ndvi_clean.to_csv(ndvi_output_csv)
+
+# Check that output csv exists
+if os.path.exists(ndvi_output_csv):
+    print('Output CSV exists')
+else:
+    print('Oh no, final CSV does not exist in the outputs directory.')
```
Comment (Tsrockross) [inline svetlana_kurakina_ndvi.py:902]: Really thorough to ensure that the paths and output folders exist.

**Comment by Tsrockross** [comment]
Really clean and concise code. Good use of comments too.

File: svetlana_kurakina_ndvi.py (hunk 2 above)
Comment (S-Kur) [reply svetlana_kurakina_ndvi.py:327]: So far I was sticking with the following approach: if I am making a new variable, I am using spaces around '=', when I specify input values for a function I do not use spaces. However, you made me thinking and I decided to go check the documentation and discussions on that. Here is waht i was able to find [Decumentation](https://www.python.org/dev/peps/pep-0008/#other-recommendations),  [Stackoverflow discussion](https://stackoverflow.com/questions/8853063/pep-8-why-no-spaces-around-in-keyword-argument-or-a-default-parameter-value). So i guess having no spaces around a default value in a function is kinda correct. I think i will ask in class or maybe post the question on canvas to make sure :)

File: svetlana_kurakina_ndvi.py (hunk 3 above)
Comment (S-Kur) [reply svetlana_kurakina_ndvi.py:781]: Thank you! I initially made a simpler loop, but it was assigning random color to my plot lines which was okay, but I wanted to be able to assign a specific color to each site. Thus I ended up wiht this solution, but it took me some googling and asking for advice from a person more experienced with plotting in python.

File: svetlana_kurakina_ndvi.py (hunk 4 above)
Comment (S-Kur) [reply svetlana_kurakina_ndvi.py:902]: It is probably unnessesary here, but I liked this approach when we did something like that in one of the earlier homeworks...

**Pull request closed without merging**